registrations/views.py

```diff
@@ -177,3 +177,30 @@ def post(self, request, *args, **kwargs):
         pull_third_party_registrations.apply_async(args=[self.request.user.id])
         resp = {"third_party_registration_pull_initiated": True}
         return Response(resp, status=status)
+
+
+class AddRegistrationView(APIView):
+
+    """ ThirdPartyRegistrationView Interaction
+        POST - Validates and Saves the registration
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+        status = 201
+        resp = {"registration_added": True}
+
+        source = Source.objects.get(user=self.request.user.id)
```

Comment:
I opted for not creating the `source` automatically like we do in other places, we might not know the `authority` of the `source` in the code so it would have to be created manually.